docs/src/cookbooks/hierarchy.md

```diff
@@ -167,7 +170,92 @@ val chiselCircuit = (new chisel3.stage.phases.Elaborate)
 println("```")
 ```
 
-## How do I look up parameters from a Definition, if I don't want to instantiate it?
+## How do I make case classes containing Data or Modules accessible from an instance?
+
+For case classes containing `Data`, `BaseModule`, `MemBase` or `Instance` types, you can provide an implementation of the `Lookupable` typeclass.
+
+**Note that Lookupable for Modules is deprecated, please cast to Instance instead (with `.toInstance`).**
+
+Consider the following case class:
+
+```scala mdoc:reset
+import chisel3._
+import chisel3.experimental.hierarchy.{Definition, Instance, instantiable, public}
+
+case class UserDefinedType[T <: Data, M <: RawModule](name: String, data: T, inst: Instance[M])
+```
+
+By default, instances of `UserDefinedType` will not be accessible from instances:
+
+```scala mdoc:fail
+class ChildModule extends Module {
+  @public val wire = Wire(UInt(8.W))
+}
+@instantiable
+class HasUserDefinedType extends Module {
+  val inst = Module(new ChildModule)
+  val wire = Wire(UInt(8.W))
+  @public val x = UserDefinedType("foo", wire, inst.toInstance)
+}
+```
+
+We can implement the `Lookupable` type class for `UserDefinedType` in order to make it accessible.
```

Comment:
nit 
this is getting a bit into 'explanation' over 'cookbook'

Reply:
Improved a bit I think!